pkg/systemtemplate/template.go

```diff
@@ -124,62 +124,16 @@ spec:
       labels:
         app: cattle-cluster-agent
     spec:
+      # Affinity added if user defined, otherwise add default
       affinity:
-        podAntiAffinity:
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 100
-            podAffinityTerm:
-              labelSelector:
-                matchExpressions:
-                - key: app
-                  operator: In
-                  values:
-                  - cattle-cluster-agent
-              topologyKey: kubernetes.io/hostname
-        nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-              - matchExpressions:
-                - key: beta.kubernetes.io/os
-                  operator: NotIn
-                  values:
-                    - windows
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - preference:
-              matchExpressions:
-              - key: node-role.kubernetes.io/controlplane
-                operator: In
-                values:
-                - "true"
-            weight: 100
-          - preference:
-              matchExpressions:
-              - key: node-role.kubernetes.io/control-plane
-                operator: In
-                values:
-                - "true"
-            weight: 100
-          - preference:
-              matchExpressions:
-              - key: node-role.kubernetes.io/master
-                operator: In
-                values:
-                - "true"
-            weight: 100
-          - preference:
-              matchExpressions:
-              - key: cattle.io/cluster-agent
-                operator: In
-                values:
-                - "true"
-            weight: 1
+{{ .Affinity | indent 8 }}
       serviceAccountName: cattle
       tolerations:
       {{- if .Tolerations }}
-      # Tolerations added based on found taints on controlplane nodes
+      # Tolerations added based on found taints on controlplane nodes user defined tolerations
 {{ .Tolerations | indent 6 }}
       {{- else }}
-      # No taints or no controlplane nodes found, added defaults
+      # No taints or controlplane nodes found and no user defined tolerations, add defaults
```

Comment:
This is not how it should work.
If we don't have tolerations calculated based on CP nodes, the resulting tolerations should be the default listed below with AppendTolerations appended to it.

Imagine a case of EKS cluster. If we don't specify any AppendTolerations we will end up with the defaults listed below. Now if there are any AppendTolerations added, we will get just them instead of defaults + them. Not good.

I'm thinking it could be a good idea to simply pass `AdditionalTolerations` to the template instead of rebuilding how we build `Tolerations` field.